contracts/src/TeleporterTokenSource.sol

```diff
@@ -37,14 +37,14 @@ import {IERC20} from "@openzeppelin/contracts@4.8.1/token/ERC20/ERC20.sol";
  * @param registered whether the destination bridge is registered
  * @param collateralNeeded the amount of tokens that must be first added as collateral,
  * through {addCollateral} calls, before tokens can be bridged to the destination token bridge.
- * @param tokenMultiplier the scaling factor for the amount of tokens to be bridged to the destination.
- * @param multiplyOnDestination whether the scaling factor is multiplied or divided when sending to the destination.
+ * @param sourceTokenDecimals the number of decimals that the source token has
+ * @param destinationTokenDecimals the number of decimals that the destination token has
  */
 struct DestinationBridgeSettings {
     bool registered;
     uint256 collateralNeeded;
-    uint256 tokenMultiplier;
-    bool multiplyOnDestination;
+    uint8 sourceTokenDecimals;
```

Comment:
If we do go the route of passing the explicit decimal counts all the way through to the scaling calculations, I think the source contract should have it's own single decimal count variable rather than tracking it for each destination (since it is the source for each of the destinations itself)

To be defensive, we could still include the `sourceTokenDecimal` in the `RegisterDestinationMessage`, and then require it's value to be equal to the decimals value of the source contract itself. That way, if an incorrect `sourceTokenDecimals` value is provided to a destination contract, it won't be able to register itself with the source contract.